tools/ci_connector_ops/ci_connector_ops/pipelines/pipelines/metadata.py

```diff
@@ -28,14 +29,56 @@ async def _run(self, poetry_run_args: list) -> StepStatus:
         return await self.get_step_result(poetry_run_exec)
 
 
+def get_metadata_file_from_path(context: PipelineContext, metadata_path: Path) -> dagger.File:
+    if metadata_path.is_file() and metadata_path.name != METADATA_FILE_NAME:
+        breakpoint()
+        raise ValueError(f"The metadata file name is not {METADATA_FILE_NAME}, it is {metadata_path.name} .")
+    if metadata_path.is_dir():
+        metadata_path = metadata_path / METADATA_FILE_NAME
+    if not metadata_path.exists():
+        raise FileNotFoundError(f"{str(metadata_path)} does not exist.")
+    return context.get_repo_dir(str(metadata_path.parent), include=[METADATA_FILE_NAME]).file(METADATA_FILE_NAME)
+
+
 class MetadataValidation(PoetryRun):
-    def __init__(self, context: PipelineContext, metadata_file_path: Path):
-        super().__init__(context, f"Validate {metadata_file_path}", METADATA_DIR, METADATA_LIB_MODULE_PATH)
-        self.metadata_file = self.context.get_repo_dir(str(metadata_file_path), include=["metadata.yaml"]).file("metadata.yaml")
-        self.poetry_run_container = self.poetry_run_container.with_mounted_file("metadata.yaml", self.metadata_file)
+    def __init__(self, context: PipelineContext, metadata_path: Path):
+        title = f"Validate {metadata_path}"
+        super().__init__(context, title, METADATA_DIR, METADATA_LIB_MODULE_PATH)
+        self.poetry_run_container = self.poetry_run_container.with_mounted_file(
+            METADATA_FILE_NAME, get_metadata_file_from_path(context, metadata_path)
+        )
+
+    async def _run(self) -> StepStatus:
+        return await super()._run(["metadata_service", "validate", METADATA_FILE_NAME])
+
+
+class MetadataUpload(PoetryRun):
+
+    GCS_CREDENTIALS_CONTAINER_PATH = "gcs_credentials.json"
```

Comment:
Question: Why not keep it as an env var in the container?

Reply:
No hard reason. I stored it in a constant because this string is reused a couple of time. I'm using it in ```
                "--service-account-file-path",
                self.GCS_CREDENTIALS_CONTAINER_PATH,
```
instead of using the `GOOGLE_APPLICATION_CREDENTIALS` env var because I find that explicit declaration of option is maybe more readable.